MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -231,6 +238,42 @@ public void setWebReturnToLabel(String webReturnToLabel) {
     }
 
     /**
+     * Enables Android Pay support in the {@link BuyClient}
+     * @param androidPayPublicKey The base64 encoded public key associated with Android Pay.
+     */
+    public void enableAndroidPay(String androidPayPublicKey) {
+        if (TextUtils.isEmpty(androidPayPublicKey)) {
+            throw new IllegalArgumentException("androidPayPublicKey cannot be empty");
+        }
+
+        byte[] digest;
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            digest = messageDigest.digest(androidPayPublicKey.getBytes("UTF-8"));
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+            // Do not enable Android Pay if the hash could not be computed
```

Comment:
hmm.  this should only happen on very old versions of Android.  I don't think we would want to throw a runtime exception because they should still be able to proceed with non Android Pay payment methods.  There is is nothing they can do to "recover" from the runtime exception - if the correct sha and utf8 are not supported, Android Pay will just not work.
I will log the error.  I did not want to put the burden of checking for these specific exceptions onto the developer because they should never happen.